patches/tModLoader/Terraria/Social/Steam/WorkshopBrowserModule.cs

```diff
@@ -160,5 +160,31 @@ public List<ModDownloadItem> DirectQueryItems(QueryParameters queryParams, out L
 
 		return new WorkshopHelper.QueryHelper.AQueryInstance(queryParams).QueryItemsSynchronously(out missingMods);
 	}
+
+	/// <summary>
+	/// This uses the provided modId and retrieves the list of approved mod hashes from ModBrowser.
+	/// Uses Steam Web API as a fall back when Steam API isn't available (such as headless environments)
+	/// </summary>
+	public DeveloperMetadata GetDeveloperMetadataFromModBrowser(ModPubId_t modId)
+	{
+		// Mod Doesn't Exist, return empty hashes
+		if (string.IsNullOrEmpty(modId.m_ModPubId) || string.Equals(modId.m_ModPubId, "0"))
+			return new();
+
+		if (!SteamedWraps.SteamAvailable) {
+			// If Steam Server and Steam Client are both not available, retrieve it via WebAPI
+			var itemDetails = SteamWebWrapper.GetItemMetadata(modId.m_ModPubId);
+
+			return DeveloperMetadata.Deserialize(itemDetails.Metadata);
+		}
+
+		// Mod should Exist, check Mod Browser
+		var items = DirectQueryItems(new QueryParameters() { searchModIds = [modId], queryType = QueryType.SearchDirect, returnDevMetadata = true }, out _);
+
+		if (!items.Any())
+			return new();
+
+		return items.First().DevMetadata;
```

Comment:
Can do `items.FirstOrDefault()?.DevMetadata ?? new()` if you feel like being fancy